crates/astria-conductor/src/executor/state.rs

```diff
@@ -1,60 +1,95 @@
+        Block,
+        CommitmentState,
+        GenesisInfo,
+    },
+    sequencer::v1alpha1::RollupId,
 };
+use celestia_client::celestia_types::Height as CelestiaHeight;
 use sequencer_client::tendermint::block::Height;
 
-// Maps a rollup height to a sequencer heights.
-/// # Panics
-///
-/// Panics if adding the integers overflows. Comet BFT has hopefully migrated
-/// to `uint64` heights by the times this becomes an issue.
-fn map_rollup_height_to_sequencer_height(
-    first_sequencer_height: u32,
-    current_rollup_height: u32,
-) -> Height {
-    first_sequencer_height
-        .checked_add(current_rollup_height)
-        .expect(
-            "should not overflow; if this overflows either the first sequencer height or current \
-             rollup height have been incorrectly set incorrectly set, are in the future, or the \
-             rollup/sequencer have been running for several decades without cometbft ever \
-             receiving an update to uin64 heights",
-        )
-        .into()
+#[derive(Debug)]
+pub(crate) struct State {
+    inner: Option<StateImpl>,
 }
 
 #[derive(Debug)]
-pub(crate) struct State {
+struct StateImpl {
+    genesis_info: GenesisInfo,
     commitment_state: CommitmentState,
 
     next_firm_sequencer_height: Height,
     next_soft_sequencer_height: Height,
-
-    // The sequencer height that contains the first block of the executed-upon rollup.
-    sequencer_height_with_first_rollup_block: u32,
 }
 
 impl State {
-    pub(super) fn new(
-        commitment_state: CommitmentState,
-        sequencer_height_with_first_rollup_block: u32,
-    ) -> Self {
+    pub(super) fn new() -> Self {
+        Self {
+            inner: None,
+        }
+    }
+
+    pub(super) fn init(&mut self, genesis_info: GenesisInfo, commitment_state: CommitmentState) {
+        self.inner
+            .replace(StateImpl::new(genesis_info, commitment_state));
+    }
+
+    pub(crate) fn is_init(&self) -> bool {
+        self.inner.is_some()
+    }
+
+    /// Updates the tracked state if `commitment_state` is different.
+    pub(super) fn update_if_modified(&mut self, commitment_state: CommitmentState) -> bool {
+        self.inner
+            .as_mut()
+            .expect("the state is initialized")
+            .update_if_modified(commitment_state)
+    }
+}
+
+macro_rules! forward_impls {
+    ($([$fn:ident -> $ret:ty]),*$(,)?) => {
+        impl State {
+            $(
+            pub(crate) fn $fn(&self) -> $ret {
+                self.inner
+                    .as_ref()
+                    .expect("the state is initialized")
+                    .$fn()
+            }
+            )*
+        }
+    }
+}
+
+forward_impls!(
+    [firm -> &Block],
+    [soft -> &Block],
+    [firm_parent_hash -> [u8; 32]],
+    [soft_parent_hash -> [u8; 32]],
+    [celestia_base_block_height -> CelestiaHeight],
+    [rollup_id -> RollupId],
+    [next_firm_sequencer_height -> Height],
+    [next_soft_sequencer_height -> Height],
+);
```

Comment:
what is this for / where is it used?

Reply:
This is to avoid having to write `self.inner.as_ref().expect("").<method>()` for every single method on on the inner state.